packages/dom-expressions/src/client.js

```diff
@@ -84,22 +84,36 @@ export function addEventListener(node, name, handler, delegate) {
 }
 
 export function classList(node, value, prev = {}) {
-  const classKeys = Object.keys(value),
-    prevKeys = Object.keys(prev);
+  const prevKeys = Object.keys(prev);
   let i, len;
   for (i = 0, len = prevKeys.length; i < len; i++) {
     const key = prevKeys[i];
     if (!key || key === "undefined" || key in value) continue;
     toggleClassKey(node, key, false);
     delete prev[key];
   }
```

Comment:
This loop compares prev keys that and if they are missing from the value it unsets them. I'm gathering we need to update to support objects in arrays.

Reply:
Oh mb, missed the `key in value`, I have a couple of solutions in mind. First I'm thinking the simplest way to achieve this is to just wipe them all and rebuild them. At first that sounds like it would have poor performance, but I think its unlikely developers will be setting enough classnames to make this significant, even on several components. Another option would be to track present classes with setClasses, and remove all other classes.